python/cudf_polars/cudf_polars/dsl/expressions/unary.py

```diff
@@ -47,7 +56,45 @@ def do_evaluate(
         """Evaluate this expression given a dataframe for context."""
         (child,) = self.children
         column = child.evaluate(df, context=context, mapping=mapping)
-        return Column(plc.unary.cast(column.obj, self.dtype)).sorted_like(column)
+        if (
+            self.dtype.id() == plc.TypeId.STRING
+            or column.obj.type().id() == plc.TypeId.STRING
+        ):
+            result = self._handle_string_cast(column)
+        else:
+            result = plc.unary.cast(column.obj, self.dtype)
+        return Column(result).sorted_like(column)
```

Comment:
Hmm. I think it's definitely useful from a performance perspective to copy sortedness metadata along when casting between numeric types, since the values don't change and thus the sortedness wouldn't change. However since we don't have that guarantee going back and forth to string, we'd have to scan the result to determine the sortedness. 

IMO then we shouldn't `sorted_like` at all in the string cases and the resulting column's sortedness should be unset.

Reply:
Yes. If we're casting between numeric types then I think it is legit to copy sortedness. But to/from string we should not

Reply:
Ah wait, casting is only order preserving if:

- signedness is preserved
- for integral types, the target bit width is `>=` the source bit width

So `cast(float, unsigned)`, for example, doesn't preserve sortedness. Neither does `cast(int64, int32)`.